Source/WebCore/page/Page.cpp

```diff
@@ -290,7 +290,7 @@ static constexpr OptionSet<ActivityState> pageInitialActivityState()
     return { ActivityState::IsVisible, ActivityState::IsInWindow };
 }
 
-static Ref<Frame> createMainFrame(Page& page, PageConfiguration::ClientCreatorForMainFrame&& clientCreator, RefPtr<Frame> mainFrameOpener, FrameIdentifier identifier)
+static Ref<Frame> createMainFrame(Page& page, PageConfiguration::ClientCreatorForMainFrame&& clientCreator, RefPtr<Frame>&& mainFrameOpener, FrameIdentifier&& identifier)
```

Comment:
FrameIdentifier is OK to pass by value, this is a small type.

Reply:
This is a workaround for what we suspect is a GCC bug, though. It crashes if passed by value.

Reply:
Maybe add a FIXME comment with a link to this bug report. Otherwise, somebody is likely going to change it right back to pass by value.

Reply:
The crash is still there if I pass by value FrameIdentifier, only passing by reference both parameters fixes the issue

Reply:
But that's not a "FIX", we should not do this. Either it's a compiler bug or we're working around a WebKit bug, which we should identify.

Reply:
Ok. Fair enough. It took me quite a while but finally I managed to create a simplified test case and I have reported the bug to GCC here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115135

After seeing that I'm almost sure this is a bug in GCC and not something wrong with the WebKit code.

I can add a comment on the code pointing to the GCC bug.